app/bundles/CategoryBundle/Controller/BatchContactController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+/*
+ * @copyright   2018 Mautic Contributors. All rights reserved
+ * @author      Mautic
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace Mautic\CategoryBundle\Controller;
+
+use Mautic\CategoryBundle\Entity\Category;
+use Mautic\CoreBundle\Controller\AbstractFormController;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpKernel\Event\FilterControllerEvent;
+
+class BatchContactController extends AbstractFormController
+{
+    /**
+     * Initialize object props here to simulate constructor
+     * and make the future controller refactoring easier.
+     *
+     * @param FilterControllerEvent $event
+     */
+    public function initialize(FilterControllerEvent $event)
+    {
+        $this->actionModel   = $this->container->get('mautic.category.model.contact.action');
+        $this->categoryModel = $this->container->get('mautic.category.model.category');
+    }
+
+    /**
+     * Adds or removes categories to multiple contacts defined by contact ID.
+     *
+     * @return JsonResponse
+     */
+    public function execAction()
+    {
+        $params = $this->request->get('lead_batch');
+
+        if (isset($params['ids'])) {
```

Comment:
Because backend should work no matter from where the request comes. I really don't see why you ask.

Reply:
I am asking if it is OK not to send any response when the parameter is not set (`return new JsonResponse` is inside the if.
I think response should be send in both ways. Return type is not correct otherwise, which PHPStan will catch later / we got error 500 after moving to PHP 7+ and add return type to method itself.

Reply:
Gotcha. Will update.

Reply:
Updated